pawnShop/src/main/java/com/example/pawnShop/Controller/ProductTypeController.java

```diff
@@ -0,0 +1,48 @@
+package com.example.pawnShop.Controller;
+
+import com.example.pawnShop.Dto.Product.ProductTypeDto;
+import com.example.pawnShop.Service.Contract.ProductTypeService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/product_type")
+public class ProductTypeController {
+    private final ProductTypeService productTypeService;
+
+    @GetMapping("/get_all")
+    public ResponseEntity<List<ProductTypeDto>> getAllProductsTypes(){
+        var productsTypes = productTypeService.getAllProductsTypes();
+
+        return ResponseEntity.ok(productsTypes);
+    }
+    @GetMapping("/get_by_id")
+    public ResponseEntity<ProductTypeDto> getProductTypeById(@RequestParam UUID id){
+        var productType = productTypeService.getProductTypeById(id);
+
+        return ResponseEntity.ok(productType);
+    }
+    @PostMapping("/add")
+    public ResponseEntity<String> addProductType(@RequestParam String productTypeName){
+        var newProductTypeName = productTypeService.addProductType(productTypeName);
+
+        return ResponseEntity.ok(newProductTypeName);
+    }
+    @PostMapping("/modify")
+    public ResponseEntity<String> addProductType(@RequestBody ProductTypeDto modifiedProductType){
```

Comment:
I'm not quite sure if i understand properly, but do you use one method to save and update ? If so, separate them :
@PostMapping for saving an entirely new product to the DB and send the whole payload
@PatchMapping for UPDATING only fields of the product in the DB

Example:
 @Transactional
    public Book updateBookById(UUID productId, ProductTypeDTO productTypeDTO) {
        Optional<ProductTypeDTO> existingProduct = repository.findById(productId);

        if (existingProduct.isPresent()) {
            Book updatedProduct = existingProduct.get();

            updatedProduct.setId(book.getId());
            updatedProduct.setLink(book.getLink()); //extend the entity with a link, as each product will have an unique link
            log.info("Updating product [productId={}] to [productId={}]", productId, existingProduct.get());

            repository.save(updatedProduct);
            return updatedProduct;

        } else {
            throw new IllegalArgumentException(new IllegalArgumentException(String.format("Product has not been found in the database. productId: %s", productId)));
        }